contracts/vaults/VaultPauser/RibbonVaultPauser.sol

```diff
@@ -0,0 +1,253 @@
+
+import {
+    SafeERC20
+} from "@openzeppelin/contracts/token/ERC20/utils/SafeERC20.sol";
+import {
+    ReentrancyGuardUpgradeable
+} from "@openzeppelin/contracts-upgradeable/security/ReentrancyGuardUpgradeable.sol";
+import {
+    OwnableUpgradeable
+} from "@openzeppelin/contracts-upgradeable/access/OwnableUpgradeable.sol";
+import {
+    ERC20Upgradeable
+} from "@openzeppelin/contracts-upgradeable/token/ERC20/ERC20Upgradeable.sol";
+import {IVaultPauser} from "../../interfaces/IVaultPauser.sol";
+import {Vault} from "../../libraries/Vault.sol";
+import {IRibbonThetaVault} from "../../interfaces/IRibbonThetaVault.sol";
+import {IWETH} from "../../interfaces/IWETH.sol";
+import {RibbonVault} from "../BaseVaults/base/RibbonVault.sol";
+import {ShareMath} from "../../libraries/ShareMath.sol";
+import {
+    RibbonVaultPauserStorage
+} from "../../storage/RibbonVaultPauserStorage.sol";
+
+contract RibbonVaultPauser is
+    ReentrancyGuardUpgradeable,
+    OwnableUpgradeable,
+    IVaultPauser,
+    RibbonVaultPauserStorage
+{
+    using SafeERC20 for IERC20;
+    using SafeMath for uint256;
+
+    /************************************************
+     *  NON UPGRADEABLE STORAGE
+     ***********************************************/
+
+    /// @notice Stores all the vault's paused positions
+    struct PauseReceipt {
+        uint16 round;
+        address account;
+        uint128 shares;
+    }
+
+    mapping(address => mapping(address => PauseReceipt[]))
+        public pausedPositions;
+
+    /************************************************
+     *  IMMUTABLES & CONSTANTS
+     ***********************************************/
+    /// @notice WETH9 0xC02aaA39b223FE8D0A0e5C4F27eAD9083C756Cc2
+    address public immutable WETH;
+    address public immutable STETH;
+
+    /************************************************
+     *  EVENTS
+     ***********************************************/
+
+    event Pause(
+        address indexed account,
+        address indexed vaultAddress,
+        uint256 share,
+        uint256 round
+    );
+
+    event Resume(
+        address indexed account,
+        address indexed vaultAddress,
+        uint256 withdrawAmount
+    );
+
+    event ProcessWithdrawal(address indexed vaultAddress, uint256 round);
+
+    /************************************************
+     *  CONSTRUCTOR & INITIALIZATION
+     ***********************************************/
+
+    /**
+     * @notice Initializes the contract with immutable variables
+     */
+    constructor(address _weth, address _steth) {
+        require(_weth != address(0), "!_weth");
+
+        WETH = _weth;
+        STETH = _steth;
+    }
+
+    // /**
+    //  * @notice Initializes the contract with storage variables.
+    //  */
+    function initialize(address _owner, address _keeper) external initializer {
+        __ReentrancyGuard_init();
+        __Ownable_init();
+        transferOwnership(_owner);
```

Comment:
We don't need to use the proxy contract pattern for the VaultPauser contract, can just use a normal constructor